src/bits.rs

```diff
@@ -0,0 +1,723 @@
+/*  Copyright 2017-2018 the Conwayste Developers.
+ *
+ *  This file is part of libconway.
+ *
+ *  libconway is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  libconway is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with libconway.  If not, see <http://www.gnu.org/licenses/>. */
+
+use std::ops::{Index, IndexMut};
+use std::cmp;
+use universe::Region;
+use rle::Pattern;
+
+
+#[derive(Eq, PartialEq, Debug, Clone, Copy)]
+pub enum BitOperation {
+    Clear,
+    Set,
+    Toggle
+}
+
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct BitGrid(pub Vec<Vec<u64>>);
+
+impl BitGrid {
+    pub fn new(width_in_words: usize, height: usize) -> Self {
+        assert!(width_in_words != 0);
+        assert!(height != 0);
+
+        let mut rows: Vec<Vec<u64>> = Vec::with_capacity(height);
+        for _ in 0 .. height {
+            let row: Vec<u64> = vec![0; width_in_words];
+            rows.push(row);
+        }
+        BitGrid(rows)
+    }
+
+    pub fn width_in_words(&self) -> usize {
+        if self.height() > 0 {
+            self.0[0].len()
+        } else {
+            0
+        }
+    }
+
+    #[inline]
+    pub fn modify_bits_in_word(&mut self, row: usize, word_col: usize, mask: u64, op: BitOperation) {
+        match op {
+            BitOperation::Set    => self[row][word_col] |=  mask,
+            BitOperation::Clear  => self[row][word_col] &= !mask,
+            BitOperation::Toggle => self[row][word_col] ^=  mask,
+        }
+    }
+
+    // Sets or clears a rectangle of bits. Panics if Region is out of range.
```

Comment:
Needs an additional forward slash for docs. No need to fix here unless you want to, since you want to do additional documentation updates in the next PR.

Reply:
Fixed!